src/power_grid_model/_core/power_grid_meta.py

```diff
@@ -197,11 +200,29 @@ def initialize_array(
     component_type = _str_to_component_type(component_type)
     if not isinstance(shape, tuple):
         shape = (shape,)
-    if empty:
-        return np.empty(shape=shape, dtype=power_grid_meta_data[data_type][component_type].dtype, order="C")
-    return np.full(
-        shape=shape,
-        fill_value=power_grid_meta_data[data_type][component_type].nan_scalar,
-        dtype=power_grid_meta_data[data_type][component_type].dtype,
-        order="C",
-    )
+    component_meta = power_grid_meta_data[data_type][component_type]
+    if attributes is None:
+        if empty:
+            return np.empty(shape=shape, dtype=component_meta.dtype, order="C")
+        return np.full(
+            shape=shape,
+            fill_value=component_meta.nan_scalar,
+            dtype=component_meta.dtype,
+            order="C",
+        )
+    data = {}
+    for attribute in attributes:
+        if component_meta.dtype.names is None:
+            raise PowerGridUnreachableHitError
```

Comment:
you need to `raise` an object, not a type
```suggestion
            raise PowerGridUnreachableHitError()
```